velox/exec/tests/WindowFunctionRegistryTest.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <gtest/gtest.h>
+
+#include "velox/exec/WindowFunction.h"
+#include "velox/expression/SignatureBinder.h"
+#include "velox/type/Type.h"
+
+namespace facebook::velox::exec::test {
+
+namespace {
+void registerWindowFunction(const std::string& name) {
+  std::vector<exec::FunctionSignaturePtr> signatures{
+      exec::FunctionSignatureBuilder()
+          .argumentType("BIGINT")
+          .argumentType("DOUBLE")
+          .returnType("BIGINT")
+          .build(),
+      exec::FunctionSignatureBuilder()
+          .typeVariable("T")
+          .argumentType("T")
+          .argumentType("T")
+          .returnType("T")
+          .build(),
+      exec::FunctionSignatureBuilder().returnType("DATE").build(),
+  };
+  exec::registerWindowFunction(name, std::move(signatures), nullptr);
+}
+} // namespace
+
+class WindowFunctionRegistryTest : public testing::Test {
+ public:
+  WindowFunctionRegistryTest() {
+    registerWindowFunction("window_func");
+    registerWindowFunction("window_Func_Alias");
+  }
+
+  void checkEqual(const TypePtr& actual, const TypePtr& expected) {
```

Comment:
this is a small function that's used only in one place; it might be more readable to fold it into the caller